src/backend/aspen/api/views/phylo_trees.py

```diff
@@ -28,9 +36,28 @@ async def get_single_phylo_tree(
     db: AsyncSession = Depends(get_db),
     az: AuthZSession = Depends(get_authz_session),
     pathogen: Pathogen = Depends(get_pathogen),
+    splitio: SplitClient = Depends(get_splitio),
 ) -> JSONResponse:
+    # get public repository for a given pathogen
+
+    preferred_public_db = splitio.get_pathogen_treatment(
+        "PATHOGEN_public_repository", pathogen
+    )
+    # get the pathogen_repo_config  for given public_repository and pathogen
+    pathogen_repo_config = await get_pathogen_repo_config_for_pathogen(
+        pathogen, preferred_public_db, db
+    )
+    if pathogen_repo_config is None:
+        raise ex.ServerException(
+            "no public repository found for given pathogen public repository"
+        )
```

Comment:
I think you can put this code in a function and use it as a fastapi dependency instead of duplicating code!

Reply:
that's a good idea